### PR TITLE
Chronos : fix check and repair of time interval

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

We found some errors in check and automatic repair of time interval, this PR will  fix these bugs.

At the same time, set `repair=True` is a little dangerous and may cause some hard to find bugs, so we will set `repair=False` and warn users call `repair=True` if there exists any low quality data.

### 1. Why the change?

We found some errors in check and automatic repair of time interval, this PR will  fix these bugs.

### 2. User API changes
Set  `repair=False`  for `from_pandas` / `from_parquet` / `from_prometheus`
```python
def from_pandas(df,
                dt_col,
                target_col,
                id_col=None,
                extra_feature_col=None,
                with_split=False,
                val_ratio=0,
                test_ratio=0.1,
                repair=False):
```
```python
def from_parquet(path,
                 dt_col,
                 target_col,
                 id_col=None,
                 extra_feature_col=None,
                 with_split=False,
                 val_ratio=0,
                 test_ratio=0.1,
                 repair=False,
                 **kwargs):
```
```python
def from_prometheus(prometheus_url,
                    query,
                    starttime,
                    endtime,
                    step,
                    target_col=None,
                    id_col=None,
                    extra_feature_col=None,
                    with_split=False,
                    val_ratio=0,
                    test_ratio=0.1,
                    repair=False,
                    **kwargs):
```

### 3. Summary of the change 

- [x] fix check and repair to support multi-id dataframe
- [x] fix check and repair to support empty dataframe
- [x] set repair=False when `from_pandas` / `from_parquet` / `from_prometheus`

### 4. How to test?
- [x] Unit test
- [x] Application test

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py [hunk 1]
```diff
@@ -89,41 +95,87 @@ def _timestamp_type_repair(df, dt_col):
...
+        from bigdl.chronos.data.utils.resample import resample_timeseries_dataframe
+        # TODO: how to change into inplace modification
+        try:
+            df = resample_timeseries_dataframe(df, dt_col=dt_col,
+                                               interval=mode,
+                                               id_col=id_col)
+            logging.warning(f"Dataframe has be resampled according to interval {mode}.")
+            return df, True
+        except:
+            return df, False
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:178]: please use `df.groupby(...).apply(resample_timeseries_dataframe, ...)` here.

Comment (rnwang04) [reply python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:180]: Have change to df.groupby.
Acceleration looks like:
image

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py [hunk 2]
```diff
@@ -89,41 +95,87 @@ def _timestamp_type_repair(df, dt_col):
...
+    if id_col is not None and len(_id_list) > 1:
+        flag = True
+        intervals = {}
+        for id_ in _id_list:
+            df_id = df[df[id_col] == id_]
+            df_column = df_id[dt_col]
+            interval = df_column.shift(-1) - df_column
+            unique_intervals = interval[:-1].unique()
+            intervals[id_] = interval
+            if len(unique_intervals) > 1:
```
[the remark this answers is not in the document]
Comment (rnwang04) [reply python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:122]: done

**Review by TheaperDeng — APPROVED** [review]
LGTM

**Pull request merged**